File: javascript/nth-prime/nth-prime.js
'use strict';
module.exports = class {
  static nth(n) {
    if (n < 1)
      throw new Error('Prime is not possible');
    let primes = [2, 3, 5];
    let i = primes[primes.length - 1] + 1;
    while (primes.length < n) {
      let isPrime = true;
      for (let j = 0; j < primes.length; j++) {
        if (i % primes[j] === 0) {
          isPrime = false;
          break;
        }
      }
      if (isPrime)
        primes.push(i);
      i += 1;
    }
    return primes[n - 1];
  }
};

File: ruby/nth-prime/nth_prime.rb
# :nodoc
class Prime
  # rubocop:disable Style/PredicateName
  def self.is_prime(primes, i)
    (0..(primes.length - 1)).each do |j|
      return false if (i % primes[j]).zero?
    end
    true
  end

  def self.nth(n)
    raise ArgumentError, 'Prime is not possible' if n < 1
    primes = [2, 3, 5]
    i = 6
    while primes.length < n
      primes.push(i) if is_prime(primes, i)
      i += 1
    end
    primes[n - 1]
  end
end
